### ISSUE TITLE
MQTT entities naming for Home Assistant guidelines

### ISSUE THREAD

**Issue by xorguy** [rand256/valetudo#506]
**Describe the bug**

Starting with Home Assistant 2023.8.0 a warning about entities naming that will stop working in upcoming release (2024.2).

More information [The naming of MQTT entities changes to correspond with HA guidelines](https://developers.home-assistant.io/blog/2023-057-21-change-naming-mqtt-entities/)

**How to Reproduce**

1. Install Home Assistant 2023.8 or greater with MQTT'...'
2. Setup MQTT at vacuum
3. Discover and add device to MQTT
4. See warning at Home Assistant settings page

**Expected behavior**

Devices are discovered and no warning is shown.

**Comment by rand256** [comment]
i can take a look into this one but i'm not going to install Home Assistant or anything, so maybe could you at least show the warnings you're talking about? or is it like "your entries are wrong but we won't tell you which ones and how"?

**Comment by dckiller51** [comment]
> i can take a look into this one but i'm not going to install Home Assistant or anything, so maybe could you at least show the warnings you're talking about? or is it like "your entries are wrong but we won't tell you which ones and how"?

Hello,

```
		this.autoconf_payloads = {
			vacuum: {
				name: this.identifier,
				unique_id: this.identifier,
				schema: "state",
				device: { manufacturer: "Roborock", model: this.vacuum.model, name: this.identifier, identifiers: [ this.identifier ], sw_version: valetudo.version },
				supported_features: [
					"start",
					"pause",
					"stop",
					"return_home",
					"battery",
					"status",
					"locate",
					"clean_spot",
					"fan_speed",
					"send_command"
				],
```
to

```
		this.autoconf_payloads = {
			vacuum: {
				name: this.identifier,
				unique_id: this.identifier,
				schema: "state",
				device: { manufacturer: "Roborock", model: this.vacuum.model, name: "vacuum", identifiers: [ this.identifier ], sw_version: valetudo.version },
				supported_features: [
					"start",
					"pause",
					"stop",
					"return_home",
					"battery",
					"status",
					"locate",
					"clean_spot",
					"fan_speed",
					"send_command"
				],
```
I think that should fix the problem.

Discussion link [https://community.home-assistant.io/t/psa-mqtt-name-changes-in-2023-8/598099](url)

for valetudo-mapper

```
    this.autoconf_payloads = {
        map: {
            name: this.identifier + "_map",
            unique_id: this.identifier + "_map",
            device: { manufacturer: "Roborock", name: this.identifier, identifiers: [ this.identifier ] },
            topic: this.topics.map
        }
    };
```
to
```
    this.autoconf_payloads = {
        map: {
            name: this.identifier,
            unique_id: this.identifier + "_map",
            device: { manufacturer: "Roborock", name: "map", identifiers: [ this.identifier ] },
            topic: this.topics.map
        }
    };
```

**Comment by xorguy** [comment]
These are the warning messages.

<img width="582" alt="Captura de pantalla 2023-08-06 a las 18 44 47" src="https://github.com/rand256/valetudo/assets/6990151/b31438fb-abc6-4bc0-ab7f-2a0dbf65302a">
<img width="588" alt="Captura de pantalla 2023-08-06 a las 18 45 06" src="https://github.com/rand256/valetudo/assets/6990151/0df1a84d-0f7d-497f-9119-6b8203bb8c9a">

**Comment by MadJoker0815** [comment]
@dckiller51 
This really does fix it for HomeAssistant.
I changed the line in valetudo-mapper and the warning disapears after a Mapper and HA restart.
My guess is that it will be the same with valetudo itself.

**Comment by dekiesel** [comment]
Is there anything I can help with to get this merged? It would be great to have clean logs.

**Comment by rand256** [comment]
i've updated the mapper today, it was autopublished to docker hub few hour ago.

as for the vacuum itself, could anyone please check if [this new binary](https://gv.erinome.net/valetudo-20230915.tar.gz) fixes the naming issue before i actually publish it along with prebuild images and other stuff? i'm pretty sure it should, but still. i myself don't use home assistant and it seems i've even been banned recently from cloudmqtt service i used for testing before (it just doesn't allow me to connect now), so it's getting a little difficult to test all these things.

**Comment by rand256** [comment]
well. i have just released version 0.10.10 which includes the aforementioned changes (and some more).
still i'm not sure if they work properly since no one has checked them and i don't have HA, but let's assume they will.

**Comment by xorguy** [comment]
Hi, I missed the comment about the new binary before the release. I've updated two vacuums with 0.10.10 and I see no complaints at Home Assistant 2023.9.2. Tried to change name and it was changed at Home Assistant immediately, so far seems good.

Thanks for the work.
